### PR TITLE
create endpoints for update and delete

### PR DESCRIPTION
**Pull request by Walaa-Volidis**
create endpoints for update and delete

### CODE REVIEW

File: src/app/api/deleteTask/[id]/route.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  { params }: { params: { id: string } }
+) {
+  try {
+    await prisma.task.delete({
+      where: {
+        id: parseInt(params.id, 10),
+      },
+    });
+
+    return new Response(JSON.stringify("this task deleted successfully"), {
```
Comment (nmaswood) [inline src/app/api/deleteTask/[id]/route.ts:14]:
```suggestion
  return NextResponse.json({message: "Task successfully deleted" });
```

File: src/app/api/deleteTask/[id]/route.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      status: 200,
+    });
+  } catch (error) {
+    return new Response(JSON.stringify({ error: (error as Error).message }), {
+      status: 400,
```
Comment (nmaswood) [inline src/app/api/deleteTask/[id]/route.ts:19]: **Nit: Generally a status code of 400 means an invalid input, 500 means unidentified server error**

File: src/app/api/deleteTask/[id]/route.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      where: {
+        id: parseInt(params.id, 10),
+      },
+    });
+
+    return new Response(JSON.stringify("this task deleted successfully"), {
+      status: 200,
+    });
+  } catch (error) {
+    return new Response(JSON.stringify({ error: (error as Error).message }), {
```
Comment (nmaswood) [inline src/app/api/deleteTask/[id]/route.ts:18]: **Suggestion: sometimes you don't want to return the actual error to the user. "An error happened" or something vague is maybe better**

Comment (nmaswood) [reply src/app/api/deleteTask/[id]/route.ts:18]: **Suggestion: better to log the error as well, if you are going to try/catch the error.**

**Question: do you ever need the try/catch? Why not just omit it?**

File: src/app/api/updateTask/[id]/route.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  userId: z.string(),
+});
+export async function PATCH(
+  request: Request,
+  { params }: { params: { id: string } }
+) {
+  try {
+    const formData = await request.formData();
+    const task = ZTaskSchema.parse({
+      title: formData.get("title") as string,
```
Comment (nmaswood) [inline src/app/api/updateTask/[id]/route.ts:19]: **Question: remove as string cast**

Comment (nmaswood) [inline src/app/api/updateTask/[id]/route.ts:19]:
```suggestion
      title: formData.get("title") ,
```

File: src/app/api/updateTask/[id]/route.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    });
+    const updatedTask = await prisma.task.update({
+      where: {
+        id: parseInt(params.id, 10),
+      },
+      data: {
+        ...task,
+      },
+    });
+    return new Response(JSON.stringify(updatedTask), { status: 200 });
```
Comment (nmaswood) [inline src/app/api/updateTask/[id]/route.ts:33]:
```suggestion
    return new NextResponse.json(...)
```

File: src/app/api/updateTask/[id]/route.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      description: formData.get("description") as string,
+      category: formData.get("category") as string,
+      date: formData.get("date") as string,
+      userId: formData.get("userId") as string,
+    });
+    const updatedTask = await prisma.task.update({
+      where: {
+        id: parseInt(params.id, 10),
+      },
+      data: {
```
Comment (nmaswood) [inline src/app/api/updateTask/[id]/route.ts:29]:
```suggestion
      data: task
```

Do you need the spread here?

File: src/app/api/updateTask/[id]/route.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import prisma from "@/lib/prisma";
+import { z } from "zod";
+
+const ZTaskSchema = z.object({
+  id: z.number().optional(),
```
Comment (nmaswood) [inline src/app/api/updateTask/[id]/route.ts:5]: **Question: Can you just grab the id from the params.route do you need it  here in this schema?**

File: src/app/api/updateTask/[id]/route.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  title: z.string(),
```
Comment (nmaswood) [inline src/app/api/updateTask/[id]/route.ts:6]: **Nit: z.string().min(1)** to make sure teh strings are non empty

**Review by nmaswood — APPROVED** [review]
Great work!

File: src/app/api/updateTask/[id]/route.ts (hunk 4 above)
Comment (Walaa-Volidis) [reply src/app/api/updateTask/[id]/route.ts:19]: done

File: src/app/api/deleteTask/[id]/route.ts (hunk 3 above)
Comment (Walaa-Volidis) [reply src/app/api/deleteTask/[id]/route.ts]: done

File: src/app/api/updateTask/[id]/route.ts (hunk 4 above)
Comment (Walaa-Volidis) [reply src/app/api/updateTask/[id]/route.ts:19]: done

File: src/app/api/deleteTask/[id]/route.ts (hunk 2 above)
Comment (Walaa-Volidis) [reply src/app/api/deleteTask/[id]/route.ts]: done

File: src/app/api/deleteTask/[id]/route.ts (hunk 1 above)
Comment (Walaa-Volidis) [reply src/app/api/deleteTask/[id]/route.ts]: done

File: src/app/api/updateTask/[id]/route.ts (hunk 6 above)
Comment (Walaa-Volidis) [reply src/app/api/updateTask/[id]/route.ts:30]: I remove spread and its not necessary

File: src/app/api/updateTask/[id]/route.ts (hunk 7 above)
Comment (Walaa-Volidis) [reply src/app/api/updateTask/[id]/route.ts:6]: done

File: src/app/api/updateTask/[id]/route.ts (hunk 8 above)
Comment (Walaa-Volidis) [reply src/app/api/updateTask/[id]/route.ts:7]: done

**Pull request merged**